xmtp_db/src/encrypted_store/group.rs

```diff
@@ -574,6 +575,37 @@ impl<C: ConnectionExt> DbConnection<C> {
             }
         })
     }
+
+    /// Get conversation IDs for all conversations that require a remote commit log publish (DMs and groups where user is super admin, excluding sync groups)
+    pub fn get_conversation_ids_for_remote_log(
+        &self,
+    ) -> Result<Vec<Vec<u8>>, crate::ConnectionError> {
+        let query = dsl::groups
+            .filter(
+                dsl::conversation_type
+                    .eq(ConversationType::Dm)
+                    .or(dsl::conversation_type
+                        .eq(ConversationType::Group)
+                        .and(dsl::should_publish_commit_log.eq(true))),
+            )
+            .select(dsl::id)
+            .order(dsl::created_at_ns.asc());
+
+        self.raw_query_read(|conn| query.load::<Vec<u8>>(conn))
+    }
+
+    pub fn get_local_commit_log_cursor(
+        &self,
+        group_id: &[u8],
+    ) -> Result<Option<i64>, crate::ConnectionError> {
+        let query = local_commit_log_dsl::local_commit_log
+            .filter(local_commit_log_dsl::group_id.eq(group_id))
+            .select(local_commit_log_dsl::commit_sequence_id)
+            .order(local_commit_log_dsl::commit_sequence_id.desc())
+            .limit(1);
```

Comment:
Do you have any thoughts on using the `rowid` as the `upload_cursor`, versus the `commit_sequence_id`? It shouldn't happen in normal operation, but I'm just thinking in terms of duplicates and out-of-order entries, as well as just the semantics of `upload whatever has been newly written'. Although I guess there's something to be said for keeping it simple, I'm unsure if I'm overthinking it.

Reply:
think youre right, rowid so we dont miss anything in our local log, all gets sent to remote 👍

Reply:
Updated - a0da74c80e8e546401a606be415de2c623e9119b
with that update added logic so we dont publish when `commit_sequence_id` = 0, as per the XIP.